### PR TITLE
Schedule momentum during training

### PR DESCRIPTION
**Pull request by Puriney**
Twin-version of `LearningRateScheduler`. 
`MomentumScheduler` would set new momentum value given epoch. 

Related: 
- Make lr and momentum shared_scalars https://github.com/fchollet/keras/pull/567
- Changing Learning Rate & Momentum During Training?https://github.com/fchollet/keras/issues/888

### CODE REVIEW

File: keras/callbacks.py [hunk 1]
```diff
@@ -422,6 +422,25 @@ def on_epoch_begin(self, epoch, logs={}):
...
+            (integer, indexed from 0) and returns a new
+            momentum as output (float).
+    '''
+    def __init__(self, schedule):
+        super(MomentumScheduler, self).__init__()
+        self.schedule = schedule
+
+    def on_epoch_begin(self, epoch, logs={}):
+        assert hasattr(self.model.optimizer, 'momentum'), \
+            'Optimizer must have a "momentum" attribute.'
```
Comment (fchollet) [inline keras/callbacks.py]: This should be checked before training, at the model setting stage. Also this is not style compliant (use `if / raise ValueError` instead).

File: keras/callbacks.py [hunk 2]
```diff
@@ -422,6 +422,25 @@ def on_epoch_begin(self, epoch, logs={}):
...
+        mmtm = self.schedule(epoch)
```
Comment (fchollet) [inline keras/callbacks.py]: Don't use needed abbreviations that make code harder to read in order so save 4 characters.
